Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -415,4 +423,41 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*}
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congrEquiv (e : α ≃ β) : FreeMonoid α ≃* FreeMonoid β :=
+  MulEquiv.mk' ⟨FreeMonoid.map e.toFun, FreeMonoid.map e.invFun, fun _ => map_invFun_map_toFun_eq e,
+    fun _ => map_toFun_map_invFun_eq e⟩ (by simp [map_mul])
+
+@[to_additive (attr := simp)]
+theorem congrEquiv_one (e : α ≃ β) : congrEquiv e (1) = 1 := rfl
+
+@[to_additive (attr := simp)]
+theorem congrEquiv_of (e : α ≃ β) (a : α) : congrEquiv e (of a) = of (e.toFun a) := rfl
```

Comment:
This is a useful lemma. Just a minor nitpick on the spelling.
```suggestion
theorem congrEquiv_of (e : α ≃ β) (a : α) : congrEquiv e (of a) = of (e a) := rfl
```
Can you please also add `congrEquiv_symm_of`?

Reply:
done!